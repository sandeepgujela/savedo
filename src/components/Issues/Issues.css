.issues {
  margin-right: auto;
  margin-left: auto;
  width:95%;
  max-width:980px;
}

.issues-list{
  border-radius: 3px ;
  border-top:1px solid #e0e4e8;
  list-style-type: none;
  padding: 0;
}

.home {
  text-decoration: none;
  display: block;
  margin: 10px;
}


@media only screen and (max-width: 620px) {
  .issues{
    width:100% ;
  }
}
