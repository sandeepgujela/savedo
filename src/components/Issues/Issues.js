import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './Issues.css';
import { connect } from 'react-redux';
import { fetchIssues } from '../../state/issues/actions';

class Issues extends Component {
  componentDidMount() {
    this.props.dispatch(fetchIssues('Singh'));
  }
  render() {
    const { prop } = this.props;
    return (
      <div className={styles.Issues}>
        <span>{prop}</span>
      </div>
    );
  }
}

export { Issues as IssuesNotConnected };

Issues.propTypes = {
  prop: PropTypes.string.isRequired,
};

export default connect()(Issues);
