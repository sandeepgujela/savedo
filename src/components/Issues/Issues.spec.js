import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import Issues, { IssuesNotConnected } from './Issues';

describe('<IssuesNotConnected />', () => {
  const build = passedProps => {
    const defaultProps = {};
    return shallow(<IssuesNotConnected {...defaultProps} {...passedProps} />);
  };

  it('should render', () => {
    const dispatch = jest.fn();
    const component = build({ dispatch });
    expect(component).toMatchSnapshot();
  });

  it('should connect', () => {
    const mockStore = configureStore();
    const store = mockStore({
      data: {},
      loading: false,
    });
    const component = shallow(<Issues store={store} />);
    expect(component).toMatchSnapshot();
  });
});
