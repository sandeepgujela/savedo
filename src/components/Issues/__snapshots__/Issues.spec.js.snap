// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IssuesNotConnected /> should connect 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Issues)
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "dispatch": [Function],
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "dispatch": [Function],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<IssuesNotConnected /> should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Issues
    dispatch={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "type": "FETCH_ISSUES",
              "url": "Singh",
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span />,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "span",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span />,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
